contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,860 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+/**
+ * @title Aerodrome AMO strategy
+ * @author Origin Protocol Inc
+ */
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { SafeCast } from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+
+import { IERC20, InitializableAbstractStrategy } from "../../utils/InitializableAbstractStrategy.sol";
+import { StableMath } from "../../utils/StableMath.sol";
+
+import { ISugarHelper } from "../../interfaces/aerodrome/ISugarHelper.sol";
+import { INonfungiblePositionManager } from "../../interfaces/aerodrome/INonfungiblePositionManager.sol";
+import { ISwapRouter } from "../../interfaces/aerodrome/ISwapRouter.sol";
+import { ICLPool } from "../../interfaces/aerodrome/ICLPool.sol";
+import { ICLGauge } from "../../interfaces/aerodrome/ICLGauge.sol";
+import { IVault } from "../../interfaces/IVault.sol";
+
+import "hardhat/console.sol";
+
+contract AerodromeAMOStrategy is InitializableAbstractStrategy {
+    using StableMath for uint256;
+    using SafeERC20 for IERC20;
+    using SafeCast for uint256;
+
+    /***************************************
+            Storage slot members
+    ****************************************/
+
+    /// @notice tokenId of the liquidity position
+    uint256 public tokenId;
+    /// @dev Minimum amount of tokens the strategy would be able to withdraw from the pool.
+    ///      minimum amount of tokens are withdrawn at a 1:1 price
+    uint256 public underlyingAssets;
+    /// @notice the gauge for the corresponding Slipstream pool (clPool)
+    /// @dev can become an immutable once the gauge is created on the base main-net
+    ICLGauge public clGauge;
+    /**
+     * @notice Specifies WETH to OETHb ratio the strategy contract aims for after rebalancing 
+     * as 18 decimal point
+     * 
+     * e.g. 0.2e18 means 20% WETH 80% OETHb
+     */
+    uint256 public poolWethShare;
+    /**
+     * @notice Specifies how the target WETH share of the pool defined by the `poolWethShare` can
+     * vary from the configured value after rebalancing. Expressed as 18 decimal point
+     */
+    uint256 public poolWethShareVarianceAllowed;
+    /**
+     * Share of liquidity to remove on rebalance expressed in 18 decimal points
+     */
+    uint128 public withdrawLiquidityShare;
+    /// @dev reserved for inheritance
+    int256[50] private __reserved;
```

Comment:
I think this number is typically adjusted so that including the used storage slots, the whole thing rounds to an even number.

Reply:
great point, changed to 45